### PR TITLE
feat(framework): guilds & channels

### PR DESCRIPTION
**Pull request by twisttaan**
Updates the guild structure to be 100% with the Discord API.

### CODE REVIEW

File: packages/disploy/src/structs/managers/ChannelManager.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+			throw new Error(`Channel is not in the guild (${this.guildId})`);
+		}
+
+		switch (raw.type) {
+			case ChannelType.GuildText:
+				return new GuildTextChannel(this.app, raw);
+			case ChannelType.GuildVoice:
+				return new GuildVoiceChannel(this.app, raw);
+			default:
+				throw new Error(`Unknown channel type: ${raw.type}`);
```
Comment (suneettipirneni) [inline packages/disploy/src/structs/managers/ChannelManager.ts:34]: Should we instead fail silently here? Discord can add new channel types in the future which would make this app crash.

**Review by suneettipirneni — CHANGES_REQUESTED** [review]

File: packages/disploy/src/structs/managers/ChannelManager.ts (hunk 1 above)
Comment (twisttaan) [reply packages/disploy/src/structs/managers/ChannelManager.ts:34]: Yeah that sounds like a better idea.

File: packages/disploy/src/structs/GuildTextChannel.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	/**
+	 * The name of the channel.
+	 */
+	public name!: string;
+
+	/**
+	 * The type of the channel.
+	 */
+	public declare type: ChannelType.GuildText;
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply packages/disploy/src/structs/GuildTextChannel.ts]: resolved in https://github.com/Disploy/disploy/pull/35/commits/d794bcf60da8cccb25b27b1dbfc3893054ff5e81

File: packages/disploy/src/structs/Channel.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import { APIChannel, Routes, Snowflake } from 'discord-api-types/v10';
+import { APIChannel, ChannelType, Routes, Snowflake } from 'discord-api-types/v10';
 import type { App } from '../client';
 import { SnowflakeUtil } from '../utils';
 import { Base } from './Base';
 
-export class Channel extends Base {
+export class BaseChannel extends Base {
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply packages/disploy/src/structs/Channel.ts]: resolved in https://github.com/Disploy/disploy/pull/35/commits/d794bcf60da8cccb25b27b1dbfc3893054ff5e81

File: packages/disploy/src/structs/Guild.ts [hunk 4]
```diff
@@ -18,10 +24,118 @@ export class Guild extends Base {
 	 */
 	public name!: string;
 
+	/**
+	 * The AFK channel of the guild.
+	 */
+	public afkChannel!: ToBeFetched<BaseChannel> | null;
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply packages/disploy/src/structs/Guild.ts]: This is used throughout the code-base and is out of scope to fix in this pull request, I will later create another pull request resolving this though.

File: packages/disploy/src/structs/Channel.ts [hunk 5]
```diff
@@ -14,6 +14,11 @@ export class Channel extends Base {
 	 */
 	public createdTimestamp!: number;
 
+	/**
+	 * The type of the channel.
+	 */
+	public type!: ChannelType;
+
 	public constructor(app: App, raw: APIChannel) {
 		super(app);
```
[the remark this answers is not in the document]
Comment (twisttaan) [reply packages/disploy/src/structs/Channel.ts]: This is used throughout the code-base and is out of scope to fix in this pull request, I will later create another pull request resolving this though.

File: packages/disploy/src/structs/managers/ChannelManager.ts (hunk 1 above)
Comment (twisttaan) [reply packages/disploy/src/structs/managers/ChannelManager.ts]: resolved in https://github.com/Disploy/disploy/pull/35/commits/e4836c89c0d340b506f1c190c5a299a8401066db

Comment (twisttaan) [reply packages/disploy/src/structs/managers/ChannelManager.ts]: decided to just return an object with the type field so if the user was switching through channel types, it just falls through.

**Review by suneettipirneni — APPROVED** [review]

**Pull request merged**